src/system/temporary_directory.py

```diff
@@ -17,9 +16,21 @@
 
 def g__handleRemoveReadonly(func: FunctionType, path: str, exc: Any) -> Any:
     excvalue = exc[1]
-    if func in (os.rmdir, os.remove, os.unlink) and excvalue.errno == errno.EACCES:
-        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)  # 0777
-        func(path)
+    logging.debug(f"excvalue {excvalue}")
+    logging.debug(f"func {func.__name__}")
+    if func in (os.rmdir, os.remove, os.unlink):
+        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO | stat.S_IWRITE | stat.S_IREAD)  # 0777 nix* / +wr windows
+        try_total = 3
+        for try_count in range(try_total):  # Re-run func to force deletion especially on windows
+            try:
+                logging.debug(f'Try count: {try_count + 1}/{try_total}')
+                func_result = None
+                func_result = func(path)
+                if func_result is None:
+                    break
+            except Exception as ex:
+                logging.debug(func_result)
```

Comment:
Do `logging.warn` of the actual error inside `ex`.

I would remove `logging.debug`'s.

Reply:
changed to warn.